### PR TITLE
refactor dashboard

### PR DESCRIPTION
**Pull request by josefaidt**
*Issue #, if available:*

closes https://github.com/aws-amplify/discord-bot/issues/370
closes https://github.com/aws-amplify/discord-bot/issues/377
closes https://github.com/aws-amplify/discord-bot/issues/364

*Description of changes:*

- adds group-by helpers
- adds filter-by helpers
- adds Tag filtering
- adds two pie charts for Tags
- adds Tag health table
- adds constants to ensure filtering, grouping, and inputs use the same conventions
- adds ValueOf TypeScript helper
- refactors filtering, grouping
- refactors server load
- refactors helper directory, organization
- refactors out GitHub helpers
- refactors component directory
- modifies CSV download to include filtered dataset

localhost_3000_dashboard (2)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by cwomack — APPROVED** [review]
LGTM!

File: src/routes/dashboard/GuildMembers.svelte [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      <p class="ha-dashboard--member-description">Members Online</p>
+    </div>
+  </Column>
+</Row>
+
+<style>
+  /* .ha-dashboard--member-col {
+    padding: var(--cds-spacing-04);
+  } */
+
```
Comment (josefaidt) [inline src/routes/dashboard/GuildMembers.svelte:42]:
```suggestion
```

File: src/routes/dashboard/GuildMembers.svelte [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    </div>
+  </Column>
+</Row>
+
+<style>
+  .ha-dashboard--member-container {
+    /* display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center; */
```
Comment (josefaidt) [inline src/routes/dashboard/GuildMembers.svelte:43]:
```suggestion
```

**Review by cwomack — APPROVED** [review]
Still LGTM

**Pull request merged**